python/fledge/services/core/server.py

```diff
@@ -1620,15 +1620,28 @@ async def get_track(cls, request):
 
     @classmethod
     async def add_track(cls, request):
+
         data = await request.json()
+
         if not isinstance(data, dict):
             raise ValueError('Data payload must be a dictionary')
 
+        jsondata=data.get("data")
+
         try:
-            result = await cls._asset_tracker.add_asset_record(asset=data.get("asset"),
-                                                               plugin=data.get("plugin"),
-                                                               service=data.get("service"),
-                                                               event=data.get("event"))
+            if jsondata is None:
+
```

Comment:
We can use 
```
asset_dict = {'asset': data.get("asset"), 
                      'plugin': data.get("plugin")       
                        'service': data.get("service")
                       'event': data.get("event")
 }
if jsondata is not None:
    asset_dict['jsondata'] = data.get("data")
result = await cls._asset_tracker.add_asset_record(**asset_dict)

```

Reply:
that did not work for assettracker class .